crates/rspack_binding_api/src/native_watcher.rs

```diff
@@ -0,0 +1,184 @@
+use std::boxed::Box;
+
+use async_trait::async_trait;
+use napi::bindgen_prelude::{FnArgs, Promise};
+use napi_derive::*;
+use rspack_fs::{FsWatcher, FsWatcherOptions, Ignored, PathUpdater};
+use rspack_napi::threadsafe_function::ThreadsafeFunction;
+
+struct SafetyIgnored {
+  f: ThreadsafeFunction<String, Promise<bool>>,
+}
+
+#[async_trait]
+impl Ignored for SafetyIgnored {
+  async fn ignore(&self, path: &str) -> bool {
+    self
+      .f
+      .call_with_promise(path.to_string())
+      .await
+      .unwrap_or_default()
+  }
+}
+
+type JsCallback = ThreadsafeFunction<FnArgs<(Option<napi::Error>, Vec<String>, Vec<String>)>, ()>;
+type JsCallbackUndelayed = ThreadsafeFunction<String, ()>;
+
+#[napi(object, object_to_js = false)]
+pub struct NativeWatcherOptions {
+  pub follow_symlinks: Option<bool>,
+
+  pub poll_interval: Option<u32>,
+
+  pub aggregate_timeout: Option<u32>,
+
+  #[napi(ts_type = "(path: string) => Promise<boolean>")]
+  /// A function that will be called with the path of a file or directory that is ignored.
+  pub ignored: Option<ThreadsafeFunction<String, Promise<bool>>>,
+}
+
+#[napi]
+pub struct NativeWatcher {
+  watcher: FsWatcher,
+  close: bool,
+}
+
+#[napi]
+impl NativeWatcher {
+  #[napi(constructor)]
+  pub fn new(options: NativeWatcherOptions) -> Self {
+    let ignored = options
+      .ignored
+      .map(|f| Box::new(SafetyIgnored { f: f.clone() }) as Box<dyn Ignored>);
+
+    let watcher = FsWatcher::new(
+      FsWatcherOptions {
+        follow_symlinks: options.follow_symlinks.unwrap_or(true),
```

Comment:
Should we default to `false`, align with https://github.com/webpack/watchpack/blob/332b55016b7c32dab4134f793ca71a5141bd10c1/lib/watchpack.js#L60